buildSrc/src/main/kotlin/com/datadog/gradle/plugin/jsonschema/generator/ClassJsonElementDeserializerGenerator.kt

```diff
@@ -202,10 +194,10 @@ class ClassDeserializerGenerator(
             )
             is TypeDefinition.OneOfClass,
             is TypeDefinition.Class -> addStatement(
-                "%L.add(%T.%L(it.toString()))",
+                "%L.add(%T.%L(it as JsonObject))",
```

Comment:
If `it` is not `JsonObject` here then `ClassCastException` will be thrown which is not caught by the `catch` blocks. Alternatively we can use `.asJsonObject` call which will throw `IllegalStateException` which is caught.